### PR TITLE
Apps 6483 dash integration improvements

### PR DESCRIPTION
**Pull request by yaniv-li**

### CODE REVIEW

File: src/plugins/dash/videojs-dash.js [hunk 1]
```diff
@@ -96,94 +96,15 @@ class Html5DashJS extends Component {
...
-        } else if (event.event.match('MEDIA_ERR_ENCRYPTED')) {
-          this.player.error({ code: 5, message: event.event });
-        } else if (event.event.match('UNKNOWN')) {
-          // We shouldn't ever end up here, since this would mean a
-          // `MediaError` thrown by the video element that doesn't comply
-          // with the W3C spec. But, since we should handle the error,
-          // throwing a MEDIA_ERR_SRC_NOT_SUPPORTED is probably the
-          // most reasonable thing to do.
-          this.player.error({ code: 4, message: event.event });
+        if (event.error.code >= 10 && event.error.code <= 35) {
```
Comment (maoznir) [inline src/plugins/dash/videojs-dash.js:99]: Please add description of what each error means

File: src/plugins/dash/videojs-dash.js [hunk 2]
```diff
@@ -96,94 +96,15 @@ class Html5DashJS extends Component {
...
+          this.player.error({ code: event.error.code, dashjsErrorCode: event.error.code, message: event.error.message });
+        } else if (event.error.code >= 100 && event.error.code <= 114) {
+          this.player.error({ code: 5, dashjsErrorCode: event.error.code, message: event.error.message });
         } else {
-          // Buffer allocation error
-          this.player.error({ code: 4, message: event.event });
+          this.player.error({ code: event.error.code, message: event.error.message });
         }
-
-      } else if (event.error === 'capability' && event.event === 'encryptedmedia') {
```
Comment (maoznir) [inline src/plugins/dash/videojs-dash.js:155]: why do we remove all this? whats the effect?

File: src/plugins/dash/videojs-dash.js [hunk 3]
```diff
@@ -266,7 +187,7 @@ class Html5DashJS extends Component {
     this.mediaPlayer_.attachSource(manifestSource);
     this.tech_.triggerReady();
     this.on(this.tech_, 'seeking', () => {
-      this.mediaPlayer_.seek(this.tech_.currentTime());
+      this.mediaPlayer_.seek((this.tech_.currentTime() - 8).toFixed(2));
```
Comment (maoznir) [inline src/plugins/dash/videojs-dash.js:190]: why this change? if its needed please add comment explaining it, thanks

File: src/video-player.js [hunk 4]
```diff
@@ -495,7 +495,7 @@ class VideoPlayer extends Utils.mixin(Eventable) {
       const error = this.videojs.error();
       if (error) {
         let type = this.videojs.cloudinary.currentSourceType();
-        if ((error.code === 4 || error.code === 25) && (type === 'VideoSource' || type === 'AudioSource')) {
+        if (error.code === 4 && (type === 'VideoSource' || type === 'AudioSource')) {
```
Comment (maoznir) [inline src/video-player.js:498]: is error code 25 handled elsewhere? what does it mean? can you please add a comment explaining why code 4 is handled here this way?

File: src/plugins/dash/videojs-dash.js [hunk 5]
```diff
@@ -96,94 +96,15 @@ class Html5DashJS extends Component {
...
-      }
-      if (event.error === 'capability' && event.event === 'mediasource') {
-        // No support for MSE
-        this.player.error({
-          code: 4,
-          message: 'The media cannot be played because it requires a feature ' +
-            'that your browser does not support.'
-        });
-
-      } else if (event.error === 'manifestError' && (
```
Comment (maoznir) [inline src/plugins/dash/videojs-dash.js:109]: why is this not needed?

**Review by maoznir — CHANGES_REQUESTED** [review]

File: src/plugins/dash/videojs-dash.js (hunk 2 above)
Comment (yaniv-li) [reply src/plugins/dash/videojs-dash.js:155]: no effect this from the old implmentation

File: src/video-player.js (hunk 4 above)
Comment (yaniv-li) [reply src/video-player.js:498]: error 25 was a quick fix / workaround for a bug

File: src/plugins/dash/videojs-dash.js (hunk 5 above)
Comment (yaniv-li) [reply src/plugins/dash/videojs-dash.js:109]: old implementation

**Comment by yaniv-li** [comment]
@maoznir please review

**Review by maoznir — APPROVED** [review]
